R/as.SpatRaster.R

```diff
@@ -0,0 +1,57 @@
+#' as.SpatRaster
+#'
+#' Convert magclass object to a  object
+#'
+#'
+#' @param x MAgPIE object
+#' @param res spatial data resolution. If not provided it will be guessed.
+#' @return A  object
+#' @author Jan Philipp Dietrich
+#' @seealso \code{\link{getCoords}}
+#' @examples
+#'
+#' if (requireNamespace("raster", quietly = TRUE)) {
+#'    r <- raster::brick(ncols = 360, nrows = 180, nl = 4)
+#'    r[85:89, 176:179] <- (1:20 %*% t(1:4))
+#'    r[15:19, 76:79] <-   (10 + 1:20 %*% t(1:4))
+#'
+#'    names(r) <- c("y2000..bla", "y2001..bla", "y2000..blub", "y2001..blub")
+#'    m <- as.magpie(r)
+#'    r2 <- as.SpatRaster(m)
+#' }
+#' @export
+
+
+as.SpatRaster <- function(x, res = NULL) { # nolint
+  if (!is.magpie(x)) stop("Input is not a magpie object")
+  if (!requireNamespace("terra", quietly = TRUE)) stop("The package \"terra\" is required!")
+
+  .guessRes <- function(xy) {
+    .tmp <- function(x) {
+      return(suppressWarnings(min(diff(sort(unique(x))))))
+    }
+    guess <- min(.tmp(xy[[1]]), .tmp(xy[[2]]))
+    if (is.infinite(guess)) guess <- 0.5
```

Comment:
Can you explain in a comment what's happening here, and why 0.5 is sensible?

Reply:
historic reasons as 0.5 is our default resolution

Reply:
I meant a comment in the code :wink: Also, when/why is the guess sometimes infinite?